src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -344,6 +348,40 @@ protected final void handleSelectedKey(final SelectionKey sk) throws IOException
 
   }
 
+  private void handleFullTasksCircularBuffer(Runnable r, int sleepMillis)
+  {
+    final Thread currentThread = Thread.currentThread();
+    if (eventThread == currentThread) {
+      int size = tasks.size();
+      if (size == tasks.capacity()) {
+        try {
+          while (size > 0) {
+            tasks.pollUnsafe().run();
+            size--;
+          }
+        } catch (RuntimeException e) {
+          logger.error("", e);
+          throw e;
+        } catch (Exception e) {
+          logger.error("", e);
+          throw new RuntimeException(e);
+        }
+      } else {
+        logger.error("Failed to add Runnable {} to tasks {} and size {} is not equal to capacity {}.", r,
+            tasks, size, tasks.capacity());
+        throw new IllegalStateException("size " + size + " should be equal to capacity " + tasks.capacity());
+      }
+    } else {
+      selector.wakeup();
+      try {
+        sleep(sleepMillis);
+      } catch (InterruptedException ex) {
+        logger.error("Interrupted", ex);
```

Comment:
yeah... raise the exception. do not log it under error. netlet is a library - let the application do the logging at the levels info and above. feel free to use trace or debug if you really feel that you need to log. I would not log it also if I am raising exception.